genie-web/src/main/java/com/netflix/genie/web/dtos/ArchivedJobMetadata.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  Copyright 2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.dtos;
+
+import com.netflix.genie.common.internal.dto.DirectoryManifest;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.net.URI;
+
+/**
+ * A simple POJO for a compound value of related information to a job archived location and files.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+@Getter
+@AllArgsConstructor
+@EqualsAndHashCode(doNotUseGetters = true)
+@ToString(doNotUseGetters = true)
+public class ArchivedJobMetadata {
```

Comment:
After going through the rest of the commits in this PR, i'm not sure what is the value of exposing this, as opposed to just returning the manifest.
The job id is the key used to look this up, so it's redundant probably.
And the URL is only used by implementations to retrieve, no need to expose it.

I'm not asking to change this, it's fine. But maybe i'm missing something because this DTO seems redundant.

Reply:
The way that `JobDirectoryServerServiceImpl` is currently written needs the baseURL of wherever the job directory currently resides in order to build up the correct redirection urls. As this `ArchivedJobMetadataService` now abstracts details of where that resides it has to return it somehow or the shared code between active and archived job directory serving will break. This whole thing needs some cleanup after the code split happens.